src/Utils.ts

```diff
@@ -260,27 +260,19 @@ export namespace Utils {
     }
 
     export function currentWindowsShell(): string {
-        const is32ProcessOn64Windows: boolean = process.env.hasOwnProperty('PROCESSOR_ARCHITEW6432');
-        const system32Path: string = `${process.env.windir}\\${is32ProcessOn64Windows ? 'Sysnative' : 'System32'}`;
-        const expectedLocations: { [shell: string]: string[] } = {
-            'Command Prompt': [`${system32Path}\\cmd.exe`],
-            PowerShell: [`${system32Path}\\WindowsPowerShell\\v1.0\\powershell.exe`],
-            'WSL Bash': [`${system32Path}\\bash.exe`],
-            'Git Bash': [
-                `${process.env.ProgramW6432}\\Git\\bin\\bash.exe`,
-                `${process.env.ProgramW6432}\\Git\\usr\\bin\\bash.exe`,
-                `${process.env.ProgramFiles}\\Git\\bin\\bash.exe`,
-                `${process.env.ProgramFiles}\\Git\\usr\\bin\\bash.exe`,
-                `${process.env.LocalAppData}\\Programs\\Git\\bin\\bash.exe`
-            ]
-        };
         const currentWindowsShellPath: string = Settings.External.defaultWindowsShell();
-        for (const key in expectedLocations) {
-            if (expectedLocations[key].indexOf(currentWindowsShellPath) >= 0) {
-                return key;
+        if (currentWindowsShellPath.endsWith("cmd.exe")) {
+            return 'Command Prompt';
+        } else if (currentWindowsShellPath.endsWith("powershell.exe")) {
+            return 'PowerShell';
+        } else if (currentWindowsShellPath.endsWith("bash.exe")) {
+            if (currentWindowsShellPath.includes("Git")) {
+                return 'Git Bash';
             }
+            return 'WSL Bash';
```

Comment:
Previous logic was following VSCode. Your change looks good, and now WSL can be also launched by a wrapper named `wsl.exe`, and I think it's better to support it.

Refer to [these lines](https://github.com/Microsoft/vscode/blob/758206be09ddfaac46e941a77375ec34ed928b4b/src/vs/workbench/parts/terminal/electron-browser/terminalService.ts#L218-L238).

Reply:
I added support for `wsl.exe`. I didn't know it even existed, but it's cool because in Ubuntu WSL I set zsh.exe as my shell, and wsl.exe starts zsh.exe without the need for loading bash.